### PR TITLE
Add Rails 5 support

### PR DESCRIPTION
**Pull request by jnraine**
This makes various changes to add support for Rails 5.

- Convert all `alias_method_chain` calls to `prepend` and `super` (`alias_method_chain` was removed in Rails 5).
- Move `QueryMethods` extensions to `Relation` (after dropping `alias_method_chain`, `jit_preload?` and other methods were no longer available from `AR::Relation`).

I haven't tested this under Rails 4 – since this is a public repo, I'd like to add a travis CI support for Rails 4 and 5 using multiple Gemfiles ([like this](https://docs.travis-ci.com/user/languages/ruby/#Custom-Bundler-arguments-and-Gemfile-locations) and [this](https://github.com/thoughtbot/paperclip/blob/master/.travis.yml)).

### CODE REVIEW

File: spec/lib/jit_preloader/preloader_spec.rb [hunk 1]
```diff
@@ -55,7 +55,7 @@
           end
         end
 
-        contact_queries = [contact1, contact2, contact3].product([["addresses.count", "addresses.maximum", "phone_numbers.count"]])
+        contact_queries = [contact1, contact2, contact3].to_a.product([["addresses.count", "addresses.maximum", "phone_numbers.count"]])
```
Comment (doliveirakn) [inline spec/lib/jit_preloader/preloader_spec.rb]: i don't understand this change. is `[..]` not an array already?

Comment (jnraine) [inline spec/lib/jit_preloader/preloader_spec.rb]: Oops, yeah, I may have gone overboard on this change. In Rails 5, AR::Relation stopped delegating missing methods to the loaded collection so I had to explicitly convert any scope. I'll dial these back.

**Comment by jnraine** [comment]
@doliveirakn can you merge this? Right now our Rails 5 Gemfile is pointing to my fork and Rails 4 is pointing to your repo. It'd be nice if they were both the same. (These changes should work on Rails 4 and 5).

**Comment by doliveirakn** [comment]
@jnraine Done.

This has been released as verison 0.1.0

**Pull request merged**